ecosystem/daos/dao.py

```diff
@@ -115,13 +117,14 @@ def delete(self, repo_url: str):
         with self.storage as data:
             del data[repo_url]
 
-    def get_by_url(self, url: str) -> Repository:
+    def get_by_url(self, url: str) -> Optional[Repository]:
```

Comment:
Does this work? If so might be better than `Optional`

```suggestion
    def get_by_url(self, url: str) -> Repository | None:
```

Reply:
Yes `Optionnal` works well it's what I'm using generally. I didn't knew your option. I'll try :)